pydantic/fields.py

```diff
@@ -204,7 +212,7 @@ def __init__(
         class_validators: Optional[Dict[str, Validator]],
         model_config: Type['BaseConfig'],
         default: Any = None,
-        required: bool = True,
+        required: Union[bool, UndefinedClass] = Undefined,
```

Comment:
perhaps add a `BoolUndefined = Union[bool, UndefinedClass]` to the `TYPE_CHECKING` section above?

Reply:
Sure, done.